src/sage/libs/flint/nmod_poly_linkage.pxi

```diff
@@ -577,18 +577,37 @@ cdef inline int celement_gcd(nmod_poly_t res, nmod_poly_t a, nmod_poly_t b, unsi
         True
         sage: (G//d)*d == G
         True
+
+    Check that we catch a case where FLINT fails.  These generate the unit
+    ideal, so their GCD should be 1 (or another unit)::
+
+        sage: R.<x> = Integers(121)[]
+        sage: f = 11*x^2 + 1
+        sage: f - 61*x*f.derivative()
+        1
+        sage: gcd(f, f.derivative())
+        Traceback (most recent call last):
+        ...
+        RuntimeError: FLINT gcd calculation failed
     """
     if celement_is_zero(b, n):
         nmod_poly_set(res, a)
         return 0
 
-    # A check that the leading coefficients are invertible is *not* sufficient
+    # FLINT provides no interface for detecting errors here
     try:
         sig_on()
-        nmod_poly_gcd(res, a, b)
-        sig_off()
+        try:
+            nmod_poly_gcd(res, a, b)
+        finally:
+            sig_off()
     except RuntimeError:
-        raise ValueError("non-invertible elements encountered during GCD")
+        raise RuntimeError("FLINT gcd calculation failed") from None
```

Comment:
```suggestion
        raise RuntimeError("FLINT gcd calculation failed")
```
I think it would be better to have the full context.

Reply:
Are you really, really, really sure? I did that because the additional context you'd get is the following; I think it actively confuses the user because it makes the exception appear to originate from `sig_on()` instead of FLINT:

```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
File ~/Projects/sage/src/sage/libs/flint/nmod_poly_linkage.pxi:599, in sage.rings.polynomial.polynomial_zmod_flint.celement_gcd()
    598 try:
--> 599     sig_on()
    600     try:

RuntimeError: Aborted

During handling of the above exception, another exception occurred:
```

If you're sure, then we can do this, but I don't think it's a good idea.

Reply:
I do (in your words, really  really, really). Nearly all users encountering this will not care about where it originates (I also don't think it changing it helps say it is from FLINT), but it carries a bit extra useful information for developers.

Reply:
In that case, I concede; while I find the chained exception confusing, I'm also not a frequent Sage developer.